Mathlib/Order/KrullDimension.lean

```diff
@@ -81,9 +81,26 @@ variable {α β : Type*}
 
 variable [Preorder α] [Preorder β]
 
+/--
+Alternative definition of height, with the supremum only ranging over those series
+that end at `a`.
+-/
+lemma height_eq_iSup_last_eq (a : α) :
+    height a = ⨆ (p : LTSeries α) (_ : p.last = a), ↑(p.length) := by
+  apply le_antisymm _ (biSup_mono fun _ => le_of_eq)
+  apply iSup₂_le
+  intro p hlast
+  wlog hlenpos : p.length ≠ 0
+  · simp_all
+  -- We replace the last element in the series with `a`
+  apply le_iSup₂_of_le <|
```

Comment:
eg these uses of the `iSup₂` API show that your API for `height` is lacking

Reply:
Hmm, yes, but the reason is that I am building the basic API here. Maybe it would be cleaner to define the ”stronger” `height_le` first (which is the second part of this proof) and then build this equality next. Will try when at the computer next.

There is the tension that `height` has two similar definitions (eq or < in the condition), and some lemmas are more useful with the one form and some others with the other form (and I'd say it's helpful to only have the more useful variant)

Reply:
> (and I'd say it's helpful to only have the more useful variant)

I would say this is true only when one lemma clearly has a strictly more useful form. The universal property is not clearly more useful one way or the other since it is by definition used both forward and backward

Reply:
Right - and because I couldn't decide which of the two is more useful, and because it seems that it was unused in my development, I removed both. But will just put both back.